lib/spack/spack/solver/concretize.lp

```diff
@@ -509,6 +509,12 @@ attr("virtual_node", VirtualNode) :- attr("virtual_root", VirtualNode).
 % then the provider is the root package.
 attr("root", PackageNode) :- attr("virtual_root", VirtualNode), provider(PackageNode, VirtualNode).
 
+attr("virtual_root", VirtualNode) :- attr("root", PackageNode), provider(PackageNode, VirtualNode).
```

Comment:
This isn't needed, but seems reasonable.

Reply:
Hmm, I need to think about that. `virtual_root` was used to signal that the root spec is a virtual. Not sure in which context this new rule is helpful :thinking: